app/main/views/direct_award.py

```diff
@@ -412,3 +412,72 @@ def create_award_outcome(project_external_id, service_id):
     db.session.commit()
 
     return single_result_response("outcome", outcome), 200
+
+
+@main.route(
+    "/direct-award/projects/<int:project_external_id>/none-suitable",
+    methods=("POST",),
+    defaults={"nonawarded_reason": "none-suitable"},
+)
+@main.route(
+    "/direct-award/projects/<int:project_external_id>/cancel",
+    methods=("POST",),
+    defaults={"nonawarded_reason": "cancelled"},
+)
+def create_nonawarded_outcome(project_external_id, nonawarded_reason):
+    updater_json = validate_and_return_updater_request()
+    uniform_now = datetime.datetime.utcnow()
+
+    # an extra guard against incorrect url configuration or unintended call circumstances
+    if nonawarded_reason not in ("cancelled", "none-suitable",):
+        abort(404, f"Unknown nonawarded-reason {nonawarded_reason!r}")
```

Comment:
This feels over-the-top, we only need one of the two ways of constraining this. But feel free to keep this one and just have one `@route` with a single argument, I don't mind which we keep. It's just that both seems unnecessary?

Reply:
gone